### PR TITLE
add more fields to generated `tsconfig.json`

### PR DESCRIPTION
**Pull request by threepointone**
This PR adds some more fields to the generated `tsconfig.json` from `wrangler init`.

- `target`, `module` and `lib` are set to `esnext`, since we're comfortable supporting that on cloudflare, and esbuild handles it just fine (for now)
- `moduleResolution` is set to `node`, but of course
- some useful fields set to true: `esModuleInterop`, `allowJs`, `allowSyntheticDefaultImports`, `resolveJsonModule`
- `isolatedModules` set to true because esbuild can't handle anything else (folks can disable this if they're using `tsc` themselves, but it's such a smaaaall group of people who do so)
-  `noEmit` is `true`; we use esbuild to output code, and no one's using wrangler to generate `.d.ts` files (folks who have custom setups are welcome to do so)
- `jsx: "react"` - I think this may be the default anyway, but it's good to be explicit if people want to change it

I also pinned the version of miniflare (`^` doesn't work with non-semver versions anyway)

### CODE REVIEW

File: packages/wrangler/src/index.tsx [hunk 1]
```diff
@@ -260,9 +260,17 @@ export async function main(argv: string[]): Promise<void> {
...
-                  target: "ES2020",
-                  module: "CommonJS",
-                  lib: ["ES2020"],
+                  target: "esnext",
+                  module: "esnext",
+                  moduleResolution: "node",
+                  esModuleInterop: true,
+                  allowJs: true,
+                  allowSyntheticDefaultImports: true,
+                  isolatedModules: true,
```
Comment (threepointone) [inline packages/wrangler/src/index.tsx:269]: Maaaybe we should throw an error if esbuild is asked to compile a `.ts` file but `isolatedModules` is set to `false`. Maybe. Kicking it down the road...

Comment (petebacondarwin) [reply packages/wrangler/src/index.tsx:269]: Does eslint not error on our behalf?

Comment (threepointone) [reply packages/wrangler/src/index.tsx:269]: looks like esbuild doesn't error on this.

File: packages/wrangler/src/index.tsx [hunk 2]
```diff
@@ -216,7 +216,7 @@ export async function main(argv: string[]): Promise<void> {
           destination,
           `compatibility_date = "${compatibilityDate}"` + "\n"
         );
-        console.log(`✨ Succesfully created wrangler.toml`);
+        console.log(`✨ Successfully created wrangler.toml`);
```
Comment (petebacondarwin) [inline packages/wrangler/src/index.tsx:219]: Spell checker FTW!

File: tsconfig.json [hunk 3]
```diff
@@ -12,6 +12,5 @@
     "jsx": "react",
     "resolveJsonModule": true
   },
-  "lib": ["esnext"],
```
Comment (petebacondarwin) [inline tsconfig.json:15]: Why this change?

**Review by petebacondarwin — APPROVED** [review]

File: tsconfig.json (hunk 3 above)
Comment (threepointone) [reply tsconfig.json:15]: It's not a valid field at all, it's actually in compilerOptions (which has the same value)

Comment (petebacondarwin) [reply tsconfig.json:15]: 🤦
